### PR TITLE
Fix a mistake in the introduction of step-18.

### PR DESCRIPTION
**Pull request by bangerth**
The description of the problem is due to Ming Yang <ykevin09@gmail.com>.

### CODE REVIEW

**Comment by masterleinad** [comment]
You might want to extend the commit message by
```
Co-authored-by: Ming Yang <ykevin09@gmail.com>
```

**Comment by bangerth** [comment]
Well, that seemed wrong -- I did author it myself: The text changes are all mine, and weren't suggested by Ming Yang. He only pointed out what was wrong. Let's just go with what we have here, he is listed in the changelog after all.

File: examples/step-18/doc/intro.dox [hunk 1]
```diff
@@ -193,14 +193,29 @@ such that
...
   \\
-  &\qquad +(\mathbf{b}(\mathbf{x},t_n)-\mathbf{b}(\mathbf{x},t_{n-1}), \varphi)_{\Gamma_N}
+  &\qquad 
+  +(\mathbf{b}(\mathbf{x},t_n)-\mathbf{b}(\mathbf{x},t_{n-1}), \varphi)_{\Gamma_N}
+  +(\sigma^{n-1} \mathbf{n}, \varphi)_{\Gamma_N}
+  \\
+  &\qquad\qquad
+  \forall \varphi \in \{\mathbf{v}\in H^1(\Omega(t_{n-1}))^d: \mathbf{v}|_{\Gamma_D}=0\}.
+@f}
+Using that $\sigma^{n-1} \mathbf{n} = \mathbf{b}(\mathbf x, t_{n-1})$,
```
Comment (masterleinad) [inline examples/step-18/doc/intro.dox]: Could you clarify why this holds?

Comment (bangerth) [reply examples/step-18/doc/intro.dox]: That's just the Neumann boundary condition stated further up ahead: 
https://github.com/dealii/dealii/blob/master/examples/step-18/doc/intro.dox#L56

Comment (masterleinad) [reply examples/step-18/doc/intro.dox]: But the Neumann boundary condition is `\mathbf{n} \varepsilon(\mathbf{u}(\mathbf{x},t))` and doesn't involve `\sigma`. The only identification we make between `\mathbf{u}` and `\sigma` is `\dot\sigma = C \varepsilon (\dot{\mathbf{u}})`.

**Comment by bangerth** [comment]
@masterleinad -- I made the connection you were missing in the formula in question. Can you take another look?

**Pull request merged**